### PR TITLE
Enable sof-docs deploy in GitHub Actions

### PR DESCRIPTION
**Pull request by xiulipan**
Disable Travis to avoid conflict.
Optimize and simplify GitHub Actions
Add new deploy job for publish branch

test with my own forked repo
- deploy repo commit
https://github.com/xiulipan/xiulipan.github.io/tree/master
- Action log
https://github.com/xiulipan/sof-docs/actions/runs/518058071
- Final HTML
https://xiulipan.github.io/latest/index.html

TODO: we may need someone with admin to sof-docs to create a personal access token
and set a variable **ACTIONS_DEPLOY_KEY** as encrypted secrets for a repository. 

guide are attached:
https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
(the only needed access is **repo/public_repo**)
https://docs.github.com/en/actions/reference/encrypted-secrets

### CODE REVIEW

**Comment by xiulipan** [comment]
@lgirdwood @deb-intel Please check with my test repos if this deploy works for you.
NOTE: please move this out of **DRAFT** after you set the **encrypted-secrets**

File: .github/actions/build-sof-doc.sh
Comment (lgirdwood) [inline .github/actions/build-sof-doc.sh:10]: newline

**Review by lgirdwood — COMMENTED** [review]
@xiulipan looks good + @intelkevinputnam for technical parts.
@xiulipan the next step is to integrate https://github.com/get-woke/woke as a CI action where @deb-intel will provide the config.

**Comment by lgirdwood** [comment]
> @xiulipan looks good + @intelkevinputnam for technical parts.
> @xiulipan the next step is to integrate https://github.com/get-woke/woke as a CI action where @deb-intel will provide the config.

@xiulipan sorry, by next step I mean "next step" schedule is after this PR is merged and maybe after CNY depending on your schedule.

**Comment by intelkevinputnam** [comment]
LGTM

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:97]: apt-get update will work forever. This is bad because it's very slow. We want the wget URL to break in the future, this will remind us to remove it and be fast again.

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:106]: Don't reduce the number of steps, this makes more time consuming to see what failed. This is also completely unrelated to the "deploy" topic of this PR, at the very least it should be submitted in a different PR (if at all)

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:57]: Thanks for the `build-sof-doc.sh` script, reduces duplication and makes it possible to test locally. Please create a script for this section too so it can tested locally too.

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:53]: I think you're missing `_build_doxy`, the doxygen APIs are probably missing.

The other option is to change `_build_doxy` to `_build/doxy/`. Avoid `_build/doxygen/` because then there will be a `_build/doxygen/doxygen`

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:72]: Can you add a comment explaining  which version gets downloaded and published? The last one that was just uploaded above I'm guessing?

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:61]: How many versions does this action keeps? Any storage concern? Please short comment and/or link.

File: .github/workflows/pull-request.yml
Comment (marc-hb) [inline .github/workflows/pull-request.yml:66]: Can we rename this to "publish"? Assuming I understood what it does.

**Review by marc-hb — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:53]: Just checked and the APIs are NOT missing at https://xiulipan.github.io/latest/api/uuid-api.html. Strange.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:97]: I had some test and find apt update will be faster than wget URL workaround. So I decide to use a more simpler way.
WGET way (total 2m48s build takes 57s, setup takes1m51s)
https://github.com/xiulipan/sof-docs/runs/1783208577?check_suite_focus=true
Update way(total 1m38s build takes 54s, setup takes 54s)
https://github.com/xiulipan/sof-docs/runs/1784299347?check_suite_focus=true

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:106]: I think we do not need to spilt the setup steps if it very simple with apt update/install and pip install.
PS: could you tell why we need this hack for the lax build？

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:57]: I am not sure if this is needed as we will only use these scripts once. The local action is replace for the missing YAML anchors support I think.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:53]: The sod-docs build step used the doxygen output in its build process. We do not need them in final deploy stage.

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:97]: I'm not comparing the speed of apt-update and wget. They're both slow. I want wget because it will break in the future, then we can remove it be fast. apt-get update will be slow forever. Anyway this is off-topic, totally unrelated to deploying the docs.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:61]: This is some official action, I do not see any limitation for us as we only do deploy not that often.
https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:106]: > I think we do not need to spilt the setup steps

Let's discuss this in a different PR. Don't sneakily change my design a couple days after it was merged in a totally unrelated PR.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:66]: I used the name from Travis scripts. @deb-intel and idea?

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:57]: Did you read my comment?

**Review by marc-hb — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:97]: why it break then will be fast?
I just did some clean up to speed up my test.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:106]: I have mentioned this PR did some optimization as I found some issue during my test

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:63]: I missed the `publish` branch condition.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:57]: maybe I misunderstand your here, so you want some script to run **locally**, but not a local action?

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:68]: OK, the branch is called `publish` so my job name suggestion sounds not too far off the mark. `deploy-job` is another possibility.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:57]: but why we need a script to run locally?

Comment (marc-hb) [reply .github/workflows/pull-request.yml:57]: For the ability to test  as much as possible before git pushing of course. Just a suggestion.

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:99]: > why it break then will be fast?

I'm not parsing this sorry. 

> I just did some clean up to speed up my test.

I do this all the time and I still submit unrelated changes in different PRs. Use `git rebase -i` and push HEAD~n

(I doubt merging the steps made any significant speed difference)

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:110]: > PS: could you tell why we need this hack for the lax build？

You already asked and already answered in PR #320

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:99]: > I want wget because it will break in the future, then we can remove it and be fast again.
I am just think why will above situation happen?

If you are caring about this, I am totally OK to spilt the PR if you all think this is needed. This only a draft to test all my expected change to achieve the final goal here (move deploy from Travis to GitHub action)

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:110]: Yep, I got how the `requirements-lax.txt` is used for. 
I tried to remove the `config tweaks` during my test and the build speed will be same with `requirements.txt`
So the main purpose here is to try with a faster config here? @deb-intel @lgirdwood do we really need a duplicated test here？

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:57]: but these bash only did some file move for the deploy. To open the HTML, you only need to open the `index.html` in  `_build/html/`

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:110]: > So the main purpose here is to try with a faster config here?

Maybe: not sure what you mean by "faster config". As I wrote in #320, the purpose is explained inside the `requirements-lax.txt` file. You can also look at the PR that added this file.

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:57]: If someone changes this script, they cannot really test their change before pushing to github and they cannot look at the resulting directories either.

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:99]: Thanks, what I care about is not noising and slowing down this important PR and feature with github UI discussions, optimizations, Ubuntu workarounds and other unrelated disagreements.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:57]: Of course, this script can only be test in GitHub Action ENV. And that's why I played with it in my own forked repo first.

File: .github/workflows/pull-request.yml
Comment (xiulipan) [reply .github/workflows/pull-request.yml:112]: Thanks for the tips. I carefully read through https://github.com/thesofproject/sof-docs/pull/231 and #237 and got the answer here and add my undersetting in comments

**Comment by xiulipan** [comment]
> > @xiulipan looks good + @intelkevinputnam for technical parts.
> > @xiulipan the next step is to integrate https://github.com/get-woke/woke as a CI action where @deb-intel will provide the config.
> 
> @xiulipan sorry, by next step I mean "next step" schedule is after this PR is merged and maybe after CNY depending on your schedule.

@lgirdwood @deb-intel 
I did a quick play with some existing GitHub action: https://github.com/marketplace/actions/run-woke-with-reviewdog
check detail https://github.com/xiulipan/sof-docs/pull/1

**Comment by lgirdwood** [comment]
> @lgirdwood @deb-intel
> I did a quick play with some existing GitHub action: https://github.com/marketplace/actions/run-woke-with-reviewdog
> check detail [xiulipan#1](https://github.com/xiulipan/sof-docs/pull/1)

Looks goot to me.

**Comment by deb-intel** [comment]
@xiulipan @lgirdwood Everything looks great!! We need to add two files: .wokeignore and rules-woke.yaml. I have created them already. They need to be placed at the top level, where .gitignore is located. .wokeignore lists all files to ignore when Woke is run. rules-woke.yaml lists customized rules that overwrite the default rules. 

@xiulipan, do you want me to send the files to you via email or do you want me send them via a PR? 

IMPORTANT: In order to run the customized file, the checks process must run the following:  woke -c ./rules-woke.yaml

Please let me know if you have questions!

**Comment by deb-intel** [comment]
@xiulipan I did not know how to remove Draft mode. Sorry.

**Review by deb-intel — APPROVED** [review]
Looks great. Please see my comments.

**Comment by xiulipan** [comment]
@deb-intel Thanks for the woke configure things, Please send a PR for them first and I will do some personal test in my own repo with the woke. I will try to invite you for review at those testing PRs.

**Comment by deb-intel** [comment]
@xiulipan Done! See PR #322

**Review by lgirdwood — APPROVED** [review]

File: .github/workflows/pull-request.yml
Comment (marc-hb) [reply .github/workflows/pull-request.yml:57]: Full testing requires Github Actions. Moving a few files and directories and looking at the result can and should be done locally first, it's much more efficient. Everything that can be done locally should be done locally.

**Comment by marc-hb** [comment]
> If you are caring about this, I am totally OK to spilt the PR if you all think this is needed. This only a draft to test all my expected change to achieve the final goal here (move deploy from Travis to GitHub action)

Ping?

**Comment by xiulipan** [comment]
@deb-intel @lgirdwood I will leave this PR here as reference and start to address request from @marc-hb

**Comment by xiulipan** [comment]
@deb-intel @lgirdwood @marc-hb I spilt the PR into #323 and #324 But I think it would be better to merge this one if you think all issue is addressed. I found a dependency issue with deploy and build job name.

**Comment by lgirdwood** [comment]
> @deb-intel @lgirdwood @marc-hb I spilt the PR into #323 and #324 But I think it would be better to merge this one if you think all issue is addressed. I found a dependency issue with deploy and build job name.

I'm OK with this, but I have no real experience with GH action so will wait for approvals from @marc-hb and @intelkevinputnam before merging. @marc-hb and @intelkevinputnam please make this a priority as Xiuli will be on vacation for Chinese New Year next week.

**Comment by marc-hb** [comment]
>  I spilt the PR into #323 and #324 But I think it would be better to merge this one if you think all issue is addressed. I found a dependency issue with deploy and build job name.

Thanks for the split. The dependency issue was a one-line change, #324 is running fine now.

**Comment by xiulipan** [comment]
@lgirdwood @marc-hb close now as we will move to #324 then #323

**Pull request closed without merging**
